package/gluon-mesh-vpn-tunneldigger/files/usr/bin/tunneldigger-watchdog

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/lua
+
+function restartTunneldigger()
+	io.popen('logger -t tunneldigger-watchdog "Restarting Tunneldigger."')
+	os.execute("/etc/init.d/tunneldigger restart")
+end
+
+function readPidFile()
+	local pidFile = io.open("/var/run/tunneldigger.mesh-vpn.pid", "r")
+	local returnPid = "0"
+	if pidFile~=nil then 
+		returnPid = pidFile:read("*l") 
+		io.close(pidFile)
+	end
+	return returnPid
+end
+
+function getNeighbourCount()
+	local handleBatctl = io.popen("batctl o", 'r')
+        if handleBatctl~=nil then
+                for line in handleBatctl:lines() do
+                        if line:find('mesh%-vpn') then
+				handleBatctl.close()
+                                return 1
+                        end
+                end
+        end
+	handleBatctl.close()
+        return nil
+end
+
+local uci = require('simple-uci').cursor()
+if uci:get_bool('tunneldigger', 'mesh_vpn', 'enabled') then
+	if io.popen("pgrep tunneldigger"):read("*l") ~= readPidFile() then
```

Comment:
Did that actually happen in your tests? I suppose having a valid pidfile is necessary for the restart to work?

Reply:
Actually the init.d-script for the tunneldigger is broken:

```
root@Testknoten-MPW-TD-Watchdog-Master:/# /etc/init.d/tunneldigger restart
Stopping tunneldigger for interface mesh-vpn
[47430.044632] batman_adv: bat0: Interface deactivated: mesh-vpn
  waiting for tunneldigger to stop
[47430.051001] batman_adv: bat0: Removing interface: mesh-vpn
  tunneldigger stopped
Not starting tunneldigger - missing uuid
Starting tunneldigger on mesh-vpn
```

I wanted to look into that aswell. But one thing at a time.

Reply:
To answer your question: We tested the code by manipulating the pid-file. But I don't know if this might actually happen. 

We didn't think about the logic, just copied it from the old (b)ash implementation.

Reply:
What about the tunneldigger init script? You said in https://github.com/freifunk-gluon/gluon/pull/1735#discussion_r289643359 that it was broken.

Reply:
As you can see from the output, it behaves a little weird but does its job. The init script is not part of this PR. So we should discuss that elsewhere, maybe in an issue. I tried to understand what the script does, but didn't really get the idea behind the lines of codes. To me it just seems over engineered.

Reply:
You said it was broken. If that were true the watchdog wouldn't work, hence why I asked. I assume it works sufficiently well?

Reply:
Yes, for now it works.